mirage/ramp_generator/obs_generator.py

```diff
@@ -2257,6 +2257,10 @@ def read_parameter_file(self):
         try:
             with open(self.paramfile, 'r') as infile:
                 self.params = yaml.safe_load(infile)
+                if self.params['Inst']['instrument'].lower() == 'niriss':
```

Comment:
This looks better. But I think it would be better to move these new lines of code below the try/except block. This way any problems with check_niriss_filter won't cause a FileNotFoundError to be raised.